### PR TITLE
Changes to well reference depth

### PR DESCRIPTION
**Pull request by joakim-hove**
1. The well reference depth should *not* be updated when new connections are
   added with COMPDAT.

2. The well reference depth should be recalculated when the well is updated with
   the WELSPECS keyword.

### CODE REVIEW

**Comment by joakim-hove** [comment]
jenkins build this please

**Comment by joakim-hove** [comment]
jenkins build this please

**Comment by joakim-hove** [comment]
jenkins build this please

**Comment by joakim-hove** [comment]
jenkins build this please

**Comment by totto82** [comment]
Thanks. The code looks good and works as expected (tested on the norne test case in opm-tests)

File: src/opm/parser/eclipse/EclipseState/Schedule/KeywordHandlers.cpp [hunk 1]
```diff
@@ -153,14 +154,24 @@ namespace {
                 auto well2 = std::shared_ptr<Well>(new Well( this->getWell(name, handlerContext.currentStep)));
                 auto connections = std::shared_ptr<WellConnections>( new WellConnections( well2->getConnections()));
                 connections->loadCOMPDAT(record, handlerContext.grid, handlerContext.fieldPropsManager);
-
-                if (well2->updateConnections(connections, handlerContext.grid, handlerContext.fieldPropsManager.get_int("PVTNUM")))
-                    this->updateWell(std::move(well2), handlerContext.currentStep);
-
+                if (well2->updateConnections(connections, handlerContext.grid, handlerContext.fieldPropsManager.get_int("PVTNUM"))) {
+                    this->updateWell(well2, handlerContext.currentStep);
```
Comment (bska) [inline src/opm/parser/eclipse/EclipseState/Schedule/KeywordHandlers.cpp:158]: Did you intentionally drop the `std::move(well2)` argument here?

Comment (joakim-hove) [reply src/opm/parser/eclipse/EclipseState/Schedule/KeywordHandlers.cpp]: No that was not intentional - reinstated now.

**Comment by joakim-hove** [comment]
jenkins build this please

**Review by bska — APPROVED** [review]
Thanks for the updates.  This looks good to me.  I'll merge into master.

**Pull request merged**